pkg/awsutils/awsutils.go

```diff
@@ -845,33 +860,118 @@ func (cache *EC2InstanceMetadataCache) createENI(useCustomCfg bool, sg []*string
 			SecondaryPrivateIpAddressCount: aws.Int64(int64(needIPs)),
 		}
 	}
-	if useCustomCfg {
-		log.Info("Using a custom network config for the new ENI")
-		if len(sg) != 0 {
-			input.Groups = sg
+
+	var err error
+	var networkInterfaceID string
+	if cache.useCustomNetworking {
+		input = createENIUsingCustomCfg(sg, eniCfgSubnet, input)
+		log.Infof("Creating ENI with security groups: %v in subnet: %s", aws.StringValueSlice(input.Groups), aws.StringValue(input.SubnetId))
+
+		networkInterfaceID, err = cache.tryCreateNetworkInterface(input)
+		if err == nil {
+			return networkInterfaceID, nil
+		}
+	} else {
+		if cache.useSubnetDiscovery {
+			subnetResult, vpcErr := cache.getVpcSubnets()
+			if vpcErr != nil {
+				return "", errors.Wrap(vpcErr, "failed to get subnets in vpc")
+			}
+
+			for _, subnet := range subnetResult {
+				if *subnet.SubnetId != cache.subnetID {
+					if !validTag(subnet) {
+						continue
+					}
+				}
+				log.Infof("Creating ENI with security groups: %v in subnet: %s", aws.StringValueSlice(input.Groups), aws.StringValue(input.SubnetId))
+
+				input.SubnetId = subnet.SubnetId
+				networkInterfaceID, err = cache.tryCreateNetworkInterface(input)
+				if err == nil {
+					return networkInterfaceID, nil
+				}
+			}
 		} else {
-			log.Warnf("No custom networking security group found, will use the node's primary ENI's SG: %v", aws.StringValueSlice(input.Groups))
+			log.Info("Using same security group config as the primary interface for the new ENI")
+			networkInterfaceID, err = cache.tryCreateNetworkInterface(input)
+			if err == nil {
+				return networkInterfaceID, nil
+			}
+		}
+	}
+	return "", errors.Wrap(err, "failed to create network interface")
+}
+
+func (cache *EC2InstanceMetadataCache) getVpcSubnets() ([]*ec2.Subnet, error) {
+	describeSubnetInput := &ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("vpc-id"),
+				Values: []*string{aws.String(cache.vpcID)},
+			},
+			{
+				Name:   aws.String("availability-zone"),
+				Values: []*string{aws.String(cache.availabilityZone)},
+			},
+		},
+	}
+
+	start := time.Now()
+	subnetResult, err := cache.ec2SVC.DescribeSubnetsWithContext(context.Background(), describeSubnetInput)
```

Comment:
If error is due to missing permissions fall back to default subnet and provide a warning